expression/builtin_time_test.go

```diff
@@ -1795,15 +1795,15 @@ func (s *testEvaluatorSuite) TestMakeTime(c *C) {
 		{[]interface{}{0, 58.4, 0}, "00:58:00"},
 		{[]interface{}{0, "58.4", 0}, "00:58:00"},
 		{[]interface{}{0, 58.5, 1}, "00:59:01"},
-		{[]interface{}{0, "58.5", 1}, "00:58:01"},
+		{[]interface{}{0, "58.5", 1}, "00:59:01"},
 		{[]interface{}{0, 59.5, 1}, nil},
-		{[]interface{}{0, "59.5", 1}, "00:59:01"},
+		{[]interface{}{0, "59.5", 1}, "00:00:00"},
 		{[]interface{}{0, 1, 59.1}, "00:01:59.100000"},
 		{[]interface{}{0, 1, "59.1"}, "00:01:59.100000"},
 		{[]interface{}{0, 1, 59.5}, "00:01:59.500000"},
 		{[]interface{}{0, 1, "59.5"}, "00:01:59.500000"},
 		{[]interface{}{23.5, 1, 10}, "24:01:10"},
-		{[]interface{}{"23.5", 1, 10}, "23:01:10"},
+		{[]interface{}{"23.5", 1, 10}, "24:01:10"},
```

Comment:
Seems updated result is incompatible with MySQL's behavior:
```sql
mysql> select maketime(0, "59.5", 1);
+------------------------+
| maketime(0, "59.5", 1) |
+------------------------+
| 00:59:01               |
+------------------------+
1 row in set, 1 warning (0.00 sec)

mysql> show warnings;
+---------+------+-------------------------------------------+
| Level   | Code | Message                                   |
+---------+------+-------------------------------------------+
| Warning | 1292 | Truncated incorrect INTEGER value: '59.5' |
+---------+------+-------------------------------------------+
1 row in set (0.00 sec)

mysql> select maketime(0, "58.5", 1);
+------------------------+
| maketime(0, "58.5", 1) |
+------------------------+
| 00:58:01               |
+------------------------+
1 row in set, 1 warning (0.00 sec)

mysql> select maketime("23.5",1, 10);
+------------------------+
| maketime("23.5",1, 10) |
+------------------------+
| 23:01:10               |
+------------------------+
1 row in set, 1 warning (0.00 sec)
```

Reply:
All right. Time must be specially dealt.